packages/odata-service-inquirer/src/prompts/connectionValidator.ts

```diff
@@ -438,34 +438,40 @@ export class ConnectionValidator {
                 await this._catalogV4?.listServices();
             }
         } catch (error) {
-            // We will try the v4 catalog if v2 returns a 404 or an auth code. Try the v4 catalog with the credentials provided also
-            // as the user may not be authorized for the v2 catalog specifically.
-            if (
-                this._catalogV4 &&
-                !v4Requested &&
-                this.shouldAttemptV4Catalog((error as AxiosError).response?.status)
-            ) {
-                await this._catalogV4.listServices();
-            } else {
-                // Either the v2 or v4 catalog request failed for a specific odata version, or both failed where no odata verison was specified
-                // Do some root cause analysis to determine the end user help message
-                throw error;
-            }
+            await this.handleCatalogError(error, v4Requested);
         }
     }
 
     /**
-     * Check if we should attempt to use the v4 catalog service as a fallback.
+     * Depending on the initial error, it will run the catalog request and handle the error if it occurs.
      *
-     * @param statusCode http status code, if not provided will return false as we cannot determine the reason for v2 catalog request failure
-     * @returns true if we should attempt the v4 catalog service
+     * @param error - error
```

Comment:
Determine if a v4 catalog request should be made based on the specified error and if the error originated from a v2 or v4 catalog request. If the error originated from a v2 catalog request and it is of type 401/403/404 then we try the v4 catalog. If the v4 catalog request fails, for any reason, and the v2 catalog request failed with 401/403 we will throw that error, ultimately resulting in a basic auth prompt for the end-user. Otherwise the v4 catalog request is thrown, which may also be a 401/403, again resulting in a basic auth prompt and re-validation. If both catalogs have returned 404 nothing else can be done, the error is thrown and reported.

@param error An error returned from either a v2 or v4 catalog `listServices` request.

Reply:
Updated, thank you